### ISSUE TITLE
[Client-side caching] About the Implementation of CSC

### ISSUE THREAD

**Issue by wp973** [redis/jedis#4013]
Hi, I would like to ask a question about the implementation of CSC.

Why choose the **default mode** to implement CSC? Redis documentation describes that the default mode **costs some memory on the server side**.
```java
  private void initializeClientSideCache() {
    sendCommand(Protocol.Command.CLIENT, "TRACKING", "ON");
    String reply = getStatusCodeReply();
    if (!"OK".equals(reply)) {
      throw new JedisException("Could not enable client tracking. Reply: " + reply);
    }
  }
```

Thanks~

https://redis.io/docs/latest/develop/reference/client-side-caching/#the-redis-implementation-of-client-side-caching

**Comment by sazzad16** [comment]
We would have to start from somewhere. Right?
It would never be shipped if we were trying to support all the configuration parameters in our first CSC support release.

We welcome PRs to support different CSC configuration parameters.

**Comment by wp973** [comment]
@sazzad16 Thanks for your reply!

When we use the default mode, maybe we should limit the maximum number of tracking keys to avoid consuming too much server-side memory?

As described in redis.conf:

![image](https://github.com/user-attachments/assets/02942dee-fcee-43ec-9fef-b993a02e7eab)

**Comment by sazzad16** [comment]
> When we use the default mode, maybe we should limit the maximum number of tracking keys to avoid consuming too much server-side memory?

Yes, this is a good idea to follow for the users and maintain at the server side.

**Comment by wp973** [comment]
hi @sazzad16 , we are doing POC functional verification of CSC(using Jedis). We found that the server-side push of invalidation messages consumes a lot of CPU resources on the server side.

Related issues have been submitted to the redis repository https://github.com/redis/redis/issues/13681.

Please help pay attention to this issue. Thanks!

**Comment by wp973** [comment]
In the POC, we also encountered the problem of increasing server memory caused by **tracking.** 

![image](https://github.com/user-attachments/assets/f3971ab3-543f-490d-a84a-17e10bb36823)

Related issues: https://github.com/redis/redis/issues/9231.

**Comment by atakavci** [comment]
@wp973 initial choice for CSC implementation is to "not to opt out" from the **default** behavior on server side. But for sure we are open to discuss and explore how we can improve on Jedis side here. Could you let us know how your use case benefits from opting out ? 
I am asking since the comment [here](https://github.com/redis/redis/issues/13681#issuecomment-2526869478);

> CSC is suitable for the scenario of less write and more read.

this is kind of guiding to re-evaluate the use case you have.

let us know what you think of.
